deployment/articles-dxp/01-saml/04-setting-up-service-provider.markdown

```diff
@@ -0,0 +1,163 @@
+
+2.  To begin configuring @product@ to use SAML, you must select a SAML role for
+    @product@ and choose an entity ID. Select the *Service Provider* SAML role.
+    Choose your own entity ID. Then click *Save* and a new section entitled
+    Certificate and Private Key appears.
+
+3.  The Certificate and Private Key section is for creating a keystore for SAML.
+    Click *Create Certificate* and enter the following information:
+
+    - Your common name (your first and last name)
+    - The name of your organization
+    - The name of your organizational unit
+    - The name of your city or locality
+    - The name of your state or province
+    - The name of your country
+    - The length in days that your keystore will remain valid (how long before
+      the keystore expires)
+    - The key algorithm (RSA is the default)
+    - The key length in bits (2048 is the default)
+    - The key password
+
+    Remember that the keystore has two storage options: file system storage (the
+    default) and Documents and Media storage.By default, the certificate uses
+    the SHA256 algorithm for encryption and is fingerprinted and self-signed via
+    MD5 and SHA1. When you enter all the required information, click *Save*.
+
+4.  After you clicked *Save*, check that you can view information about your
+    certificate or download your certificate. If you can, you successfully
+    created a keystore. After you create a keystore, additional options
+    appear. There are three tabs:
+
+    **General**: Enables or disables SAML SP and manages the required
+    keystore.
+ 
+    **Service Provider**: This tab manages basic and advanced configurations for
+    the SP.
+ 
+    **Identity Provider Connection**: This tab manages connections to the IdP.
+    There can be multiple IdP connections.
+
+5.  You can also generate an encryption certificate. This is a separate key for
+    encrypting assertions. If you want assertions encrypted, you must generate
+    a key for this. The procedure is exactly the same as generating your
+    certificate in step 3 above. 
+
+5.  Next, you must configure an Identity Provider connection. Click on the
+    *Identity Provider Connections* tab. Enter a name for the Identity Provider,
+    enter its entity ID, and enter its metadata URL. If you have already
+    followed the previous instructions and configured a separate @product@
+    installation as an Identify provider, you'd enter the following information:
+
+    - Name: *Liferay IdP*
+    - Entity ID: [ID of IdP]
+	- Clock Skew: Set the tolerance in milliseconds between SP and IdP.
+	- Force Authn: Whether the IdP should force reauthentication regardless of
+      context.
+    - Metadata URL: http://localhost:8080/c/portal/saml/metadata (test this URL
+      first)
+	- Name Identifier Format: See settings. 
+	- Attribute Mapping: See settings. 
+	- Keep Alive URL: See settings. 
+
+    **Important**: The Liferay SAML 3.0 Provider app supports using *either*
```

Comment:
**Change**: there is no such app "Liferay SAML 3.0". The name of the app is **"Liferay Connector to SAML 2.0"**. We only upgraded to the OpenSAMLv3 library, so "3.0" refers to the OpenSAML implementation version, but the spec itself is still 2.0. There is no SAML 3.0 spec.

Reply:
I'm pretty sure I named this differently every time I saw it. :-p  Sorry for the inconsistency.